extensions/big-data-cluster/src/bigDataCluster/tree/treeNode.ts

```diff
@@ -0,0 +1,193 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import { generateGuid } from '../utils';
+
+export type TreeNodeArg = { label: string; parent?: TreeNode };
+
+export abstract class TreeNode {
+
+	private _id: string;
+	private _label: string;
+	private _parent: TreeNode;
+	private _children: TreeNode[];
+	private _isLeaf: boolean;
+
+	constructor(arg?: TreeNodeArg) {
```

Comment:
As mentioned above there's no benefit of this vs. just having private/protected fields here:
```ts
constructor(private _label: string,
private _parent: TreeNode) {
    this.resetId();
}
```

Reply:
fixed.